internal/core/v1beta1/types.go

```diff
@@ -108,6 +109,15 @@ func NewInvalidationError(code int, err error, args ...interface{}) error {
 	}
 }
 
+func ErrorBadRequest(err error) bool {
```

Comment:
Thanks, read the post. But I'm not sure if creating a new `BadRequestError` type is the thing to do here.

My understanding is that `InvalidationError` is the _umbrella_ error type and we're putting in errors and error codes into it. So `BadRequestErrorCode` is the code along with `ResourceNotFoundErrorCode` and `InvalidationUnauthorizedErrorCode`.

But I may be wrong

Reply:
Oh, I didn't see the existing error functions returning bool's. I'm not really sure why error checking is being handled this way, but it seems odd to me. Looking more closely it seems like this might be some swagger generated code? Perhaps @dskatz can provide more context.

Reply:
This is not part of swagger.   I mostly did this to avoid cluttering with types and focus on a single struct that can be returned to the user. 

Its essentially the same approach,  as using `error.Is()`  except using a member of the custom invalidation type.

Reply:
The one important thing to note is that `InvalidationError` is special, while it implements the `error` interface can be treated like any other error.  This particular error is safe to return as JSON via an API.   

This allows us to have a separation between errors that may include unsafe or sensitive information.  I thought that defining individual types for each error that need to have the same structure to maintain the API contract was more tedious in this case.

Reply:
> The one important thing to note is that `InvalidationError` is special, while it implements the `error` interface can be treated like any other error. This particular error is safe to return as JSON via an API.

This seems fine and doesn't really seem relevant to how the error types are being raised and handled.

> This allows us to have a separation between errors that may include unsafe or sensitive information.

There should never be unsafe or sensitive information returned in an error.

> I thought that defining individual types for each error that need to have the same structure to maintain the API contract was more tedious in this case.

I think having to write special helper function for each potential http return status is far more tedious and unnecessary. Making this a special case just makes it non-standard and confusing to read.

Here is an example of how I typically would expect to work with raising and handling error types:
```
package main

import (
	"errors"
	"fmt"
	"net/http"
)

const (
  // implement specific error constants
	InvalidationErrorInvalidPath = InvalidationError("invalid path")
)

// implement error types that fulfil the error interface
type InvalidationError string

func (e InvalidationError) Error() string {
	return string(e)
}

func main() {
	// how I would expect to raise an error
	raisedError := fmt.Errorf("error: %w", InvalidationErrorInvalidPath)

	// how I would expect to handle an error
	if errors.Is(raisedError, InvalidationErrorInvalidPath) {
		fmt.Println("handled error", http.StatusOK)
	}
}
```